### PR TITLE
Feature | add a workplace | create base classes and extend on registration & add workplace flows

### PR DESCRIPTION
**Pull request by omarCreativeDev**
Create base classes for;

- `CQC question`
- `Select your workplace`
- `Select your workplace address`

and then extend and use them for add workplace and registration flows

### CODE REVIEW

**Comment by omarCreativeDev** [comment]
[(5) Parent and subsidiaries - Add a workplace - Non - CQC Regulated](https://trello.com/c/bcbbFepb/200-5-parent-and-subsidiaries-add-a-workplace-non-cqc-regulated)

File: src/app/features/add-workplace/regulated-by-cqc/regulated-by-cqc.component.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { FormBuilder } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { LocationSearchResponse } from '@core/model/location.model';
+import { LocationService } from '@core/services/location.service';
+
+@Component({
+  selector: 'app-regulated-by-cqc',
+  templateUrl: './regulated-by-cqc.component.html',
+})
+export class RegulatedByCqcComponent extends RegulatedByCQC {
```
Comment (ChazUK) [inline src/app/features/add-workplace/regulated-by-cqc/regulated-by-cqc.component.ts]: Are we making sure they are then seeing CQC Services after they select yes?

File: src/app/features/add-workplace/select-workplace/select-workplace.component.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    protected backService: BackService,
+    protected errorSummaryService: ErrorSummaryService,
+    protected formBuilder: FormBuilder,
+    protected router: Router
+  ) {
+    super(backService, errorSummaryService, formBuilder, router);
+  }
+
+  protected init(): void {
+    this.flow = '/add-workplace';
```
Comment (ChazUK) [inline src/app/features/add-workplace/select-workplace/select-workplace.component.ts]: Is this only being used in 1 place?

File: src/app/features/add-workplace/select-workplace-address/select-workplace-address.component.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    protected backService: BackService,
+    protected errorSummaryService: ErrorSummaryService,
+    protected formBuilder: FormBuilder,
+    protected router: Router
+  ) {
+    super(backService, errorSummaryService, formBuilder, router);
+  }
+
+  protected init(): void {
+    this.flow = '/add-workplace';
```
Comment (ChazUK) [inline src/app/features/add-workplace/select-workplace-address/select-workplace-address.component.ts]: Yeah I don't see why this is beneficial making this a variable as it's only used in 2 places

File: src/app/features/add-workplace/regulated-by-cqc/regulated-by-cqc.component.ts (hunk 1 above)
Comment (omarCreativeDev) [reply src/app/features/add-workplace/regulated-by-cqc/regulated-by-cqc.component.ts]: > yup but im only half way through the add workplace journey at the mo so the cqc yes route work is to follow

```
workplace-find-and-select
	regulated-by-cqc - done
	select-workplace - done
	select-workplace-address - done
	enter-workplace-address - to do
	select-main-service - to do
	confirm-workplace-details - to do
```

**Pull request merged**